pkg/handlers/office.go

```
@@ -71,3 +71,33 @@ func (h ApprovePPMHandler) Handle(params officeop.ApprovePPMParams) middleware.R
 	ppmPayload := payloadForPPMModel(*ppm)
 	return officeop.NewApprovePPMOK().WithPayload(&ppmPayload)
 }
+
+// ApproveReimbursementHandler approves a move via POST /reimbursement/{reimbursementId}/approve
+type ApproveReimbursementHandler HandlerContext
+
+// Handle ... approves a Reimbursement from a request payload
+func (h ApproveReimbursementHandler) Handle(params officeop.ApproveReimbursementParams) middleware.Responder {
+	session := auth.SessionFromRequestContext(params.HTTPRequest)
+
+	if !session.IsOfficeUser() {
+		return officeop.NewApproveReimbursementUnauthorized()
+	}
+
+	// #nosec UUID is pattern matched by swagger and will be ok
+	reimbursementID, _ := uuid.FromString(params.ReimbursementID.String())
+
+	reimbursement, err := models.FetchReimbursement(h.db, session, reimbursementID)
+	if err != nil {
+		return responseForError(h.logger, err)
+	}
+
+	reimbursement.Status = models.ReimbursementStatusAPPROVED
```

Please use the state machine methods defined on the reimbursements model for this. You should call reimbursement.Approve() instead.

You will then need to handle the error that throws. e.g. this API should return an error if you attempt to approve a reimbursement that has already been approved or paid.